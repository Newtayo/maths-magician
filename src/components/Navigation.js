import React from 'react';
import './Styles/Navigation.css';

const Navigation = () => (
  <div>
    <nav className="navigation">

      <a className="maths" href="/"> Maths Magicians</a>
      <ul className="linkholder">
        <a className="home" href="/"> Home</a>
        <a className="calculator" href="calculator"> Calculator</a>
        <a className="quote" href="addquote"> Quote</a>

      </ul>

    </nav>

  </div>
);

export default Navigation;
