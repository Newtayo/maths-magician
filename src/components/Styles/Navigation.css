.navigation {
  display: flex;
  width: 100vw;
  margin-bottom: 2%;
}

.maths {
  flex: 4;
  padding-left: 2rem;
  font-size: 3rem;
  color: black;
  text-decoration: none;
}

.linkholder {
  flex: 1;
  display: flex;
  list-style: none;
  justify-content: space-between;
  font-size: 1.2rem;
  padding-right: 10rem;
  align-items: center;
}

.calculator {
  border-left: 2px solid black;
  border-right: 2px solid black;
  padding-left: 1rem;
  padding-right: 1rem;
}
