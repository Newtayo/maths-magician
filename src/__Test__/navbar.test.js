import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import Navigation from '../components/Navigation';

test('renders correctly', () => {
  const { asFragment } = render(<Navigation />);
  expect(asFragment()).toMatchSnapshot();
});

//

test('should display the the content of the nav links', async () => {
  const { getByText } = render(<Navigation />);
  expect(getByText('Home')).toBeInTheDocument();
  expect(getByText('Calculator')).toBeInTheDocument();
  expect(getByText('Quote')).toBeInTheDocument();
});
