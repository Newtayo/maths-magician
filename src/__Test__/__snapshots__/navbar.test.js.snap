// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navigation"
      data-testid="navbar"
    >
      <a
        class="maths"
        href="/"
      >
         Maths Magicians
      </a>
      <ul
        class="linkholder"
      >
        <a
          class="home"
          href="/"
        >
           Home
        </a>
        <a
          class="calculator"
          href="calculator"
        >
           Calculator
        </a>
        <a
          class="quote"
          href="addquote"
        >
           Quote
        </a>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
